zone/corpse.cpp

```diff
@@ -1012,12 +1045,28 @@ bool Corpse::Process() {
 		corpse_graveyard_timer.Disable();
 		return false;
 	}
-	/*
-	if(corpse_res_timer.Check()) {
-		can_rez = false;
-		corpse_res_timer.Disable();
+	
+	//Player is offline. If rez timer is enabled, disable it and save corpse.
+	if (rezzable) {
+		if (!is_owner_online) {
+			if (corpse_rez_timer.Enabled()) {
+				rez_time = corpse_rez_timer.GetRemainingTime();
+				corpse_rez_timer.Disable();
+				is_corpse_changed = true;
+				Save();
+			}
+		} else { //Player is online. If rez timer is disabled, enable it.
+			if (corpse_rez_timer.Enabled()) {
+				rez_time = corpse_rez_timer.GetRemainingTime();
+			} else {
+				SetRezTimer();
+			}
+		}
+	}
```

Comment:
I called this section out before and it looks just as fishy now.

We call `Corpse::Process` several dozen frames a second. When this logic is called it is evaluated just as often, we are hitting the database several hundreds of times per second in this process loop call when a corpse is flagged as `rezzable`

Here is a gist showing corpse creation to zoning back in after death in a matter of a few seconds.

https://gist.github.com/Akkadius/cb47a0f5edbb93a9645fe661354c6392#file-gistfile1-txt-L156

Reply:
If you're running this on WF it also explains lag bombs you've had on raid wipe as this is already incredibly spammy for one toon, but would imagine it bombing the zone / SQL for several

Reply:
The Lag Bombs were not related to this, they were related to a on death item check that was loading all corpses. Similar to what PEQ deals with when checking items on zone that cause players to DC right away.

Reply:
I understand you had a separate issue with scripts, but this would certainly exasperate

Reply:
I remember, I helped you find the issue